File: app/src/main/java/com/i2medier/financialpro/model/MonthModel.java
package com.i2medier.financialpro.model;

import android.os.Parcel;
import android.os.Parcelable;
import java.text.SimpleDateFormat;
import java.util.Date;


public class MonthModel implements Parcelable {
    public static final Creator<MonthModel> CREATOR = new Creator<MonthModel>() {

        @Override
        public MonthModel createFromParcel(Parcel parcel) {
            return new MonthModel(parcel);
        }


        @Override
        public MonthModel[] newArray(int i) {
            return new MonthModel[i];
        }
    };
    public double Balance;
    public double Interest;
    public double PrincipalAmount;
    public double TaxInsPMI;
    public double TotalInterest;
    public double TotalPaid;
    public double TotalPrincipal;
    public double TotalTax;
    public Date date;
    public int month;
    public int year;

    @Override
    public int describeContents() {
        return 0;
    }

    public MonthModel() {
        this.date = new Date();
    }

    public MonthModel(double d, double d2, double d3, double d4, double d5, double d6, double d7, double d8, int i, int i2, Date date) {
        this.date = new Date();
        this.PrincipalAmount = d;
        this.Interest = d2;
        this.Balance = d3;
        this.TotalPaid = d4;
        this.TaxInsPMI = d5;
        this.TotalInterest = d6;
        this.TotalTax = d7;
        this.TotalPrincipal = d8;
        this.month = i;
        this.year = i2;
        this.date = date;
    }

    protected MonthModel(Parcel parcel) {
        this.date = new Date();
        this.PrincipalAmount = parcel.readDouble();
        this.Interest = parcel.readDouble();
        this.Balance = parcel.readDouble();
        this.TotalPaid = parcel.readDouble();
        this.TaxInsPMI = parcel.readDouble();
        this.TotalInterest = parcel.readDouble();
        this.TotalTax = parcel.readDouble();
        this.TotalPrincipal = parcel.readDouble();
        this.month = parcel.readInt();
        this.year = parcel.readInt();
    }

    public double getPrincipalAmount() {
        return this.PrincipalAmount;
    }

    public void setPrincipalAmount(double d) {
        this.PrincipalAmount = d;
    }

    public double getInterest() {
        return this.Interest;
    }

    public void setInterest(double d) {
        this.Interest = d;
    }

    public double getBalance() {
        return this.Balance;
    }

    public void setBalance(double d) {
        this.Balance = d;
    }

    public double getTotalPaid() {
        return this.TotalPaid;
    }

    public void setTotalPaid(double d) {
        this.TotalPaid = d;
    }

    public double getTaxInsPMI() {
        return this.TaxInsPMI;
    }

    public void setTaxInsPMI(double d) {
        this.TaxInsPMI = d;
    }

    public double getTotalInterest() {
        return this.TotalInterest;
    }

    public void setTotalInterest(double d) {
        this.TotalInterest = d;
    }

    public double getTotalTax() {
        return this.TotalTax;
    }

    public void setTotalTax(double d) {
        this.TotalTax = d;
    }

    public double getTotalPrincipal() {
        return this.TotalPrincipal;
    }

    public void setTotalPrincipal(double d) {
        this.TotalPrincipal = d;
    }

    public int getMonth() {
        return this.month;
    }

    public void setMonth(int i) {
        this.month = i;
    }

    public int getYear() {
        return this.year;
    }

    public void setYear(int i) {
        this.year = i;
    }

    public String getDate() {
        return new SimpleDateFormat("MMM-yyyy").format(this.date);
    }

    public Date getDateofGraph() {
        return this.date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeDouble(this.PrincipalAmount);
        parcel.writeDouble(this.Interest);
        parcel.writeDouble(this.Balance);
        parcel.writeDouble(this.TotalPaid);
        parcel.writeDouble(this.TaxInsPMI);
        parcel.writeDouble(this.TotalInterest);
        parcel.writeDouble(this.TotalTax);
        parcel.writeDouble(this.TotalPrincipal);
        parcel.writeInt(this.month);
        parcel.writeInt(this.year);
    }
}

File: app/src/main/java/com/i2medier/financialpro/model/MonthModel.kt
package com.i2medier.financialpro.model

import android.os.Parcel
import android.os.Parcelable
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class MonthModel() : Parcelable {
    var PrincipalAmount: Double = 0.0
    var Interest: Double = 0.0
    var Balance: Double = 0.0
    var TotalPaid: Double = 0.0
    var TaxInsPMI: Double = 0.0
    var TotalInterest: Double = 0.0
    var TotalTax: Double = 0.0
    var TotalPrincipal: Double = 0.0
    var month: Int = 0
    var year: Int = 0
    private var dateValue: Date = Date()

    constructor(
        principalAmount: Double,
        interest: Double,
        balance: Double,
        totalPaid: Double,
        taxInsPMI: Double,
        totalInterest: Double,
        totalTax: Double,
        totalPrincipal: Double,
        month: Int,
        year: Int,
        date: Date
    ) : this() {
        PrincipalAmount = principalAmount
        Interest = interest
        Balance = balance
        TotalPaid = totalPaid
        TaxInsPMI = taxInsPMI
        TotalInterest = totalInterest
        TotalTax = totalTax
        TotalPrincipal = totalPrincipal
        this.month = month
        this.year = year
        this.dateValue = date
    }

    private constructor(parcel: Parcel) : this() {
        PrincipalAmount = parcel.readDouble()
        Interest = parcel.readDouble()
        Balance = parcel.readDouble()
        TotalPaid = parcel.readDouble()
        TaxInsPMI = parcel.readDouble()
        TotalInterest = parcel.readDouble()
        TotalTax = parcel.readDouble()
        TotalPrincipal = parcel.readDouble()
        month = parcel.readInt()
        year = parcel.readInt()
    }

    fun getDate(): String {
        return SimpleDateFormat("MMM-yyyy", Locale.getDefault()).format(dateValue)
    }

    fun getDateofGraph(): Date {
        return dateValue
    }

    fun setDate(date: Date) {
        dateValue = date
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeDouble(PrincipalAmount)
        parcel.writeDouble(Interest)
        parcel.writeDouble(Balance)
        parcel.writeDouble(TotalPaid)
        parcel.writeDouble(TaxInsPMI)
        parcel.writeDouble(TotalInterest)
        parcel.writeDouble(TotalTax)
        parcel.writeDouble(TotalPrincipal)
        parcel.writeInt(month)
        parcel.writeInt(year)
    }

    override fun describeContents(): Int = 0

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<MonthModel> = object : Parcelable.Creator<MonthModel> {
            override fun createFromParcel(parcel: Parcel): MonthModel = MonthModel(parcel)
            override fun newArray(size: Int): Array<MonthModel?> = arrayOfNulls(size)
        }
    }
}
